ush/set_cycle_and_obs_timeinfo.py

```diff
@@ -0,0 +1,736 @@
+        An updated version of the verification configuration dictionary.
+
+        fcst_obs_matched_times_all_cycles_cumul:
+        Dictionary containing the times (in YYYYMMDDHH string format) at
+        which various field/accumlation combinations are output and at
+        which the corresponding obs type is also available.
+    """
+    # Set dictionary containing all cumulative fields (i.e. whether or not
+    # they are to be verified).  The keys are the observation types and the
+    # values are the field names in the forecasts.
+    vx_cumul_fields_all = {"CCPA": "APCP", "NOHRSC": "ASNOW"}
+
+    # Convert from datetime.timedelta objects to integers.
+    one_hour = timedelta(hours=1)
+    fcst_len_hrs = int(fcst_len/one_hour)
+    fcst_output_intvl_hrs = int(fcst_output_intvl/one_hour)
+
+    # Generate a list containing the starting times of the cycles.  This will
+    # be needed in checking that the hours-of-day of the forecast output match
+    # those of the observations.
+    cycle_start_times \
+    = set_cycle_dates(start_time_first_cycl, start_time_last_cycl, cycl_intvl,
+                      return_type='datetime')
+
+    # Initialize one of the variables that will be returned to an empty
+    # dictionary.
+    fcst_obs_matched_times_all_cycles_cumul = dict()
+
+    for obtype, field_fcst in vx_cumul_fields_all.items():
+
+        # If the current cumulative field is not in the list of fields to be
+        # verified, just skip to the next field.
+        if field_fcst not in vx_config["VX_FIELDS"]:
+            continue
+
+        # Initialize a sub-dictionary in one of the dictionaries to be returned.
+        fcst_obs_matched_times_all_cycles_cumul.update({field_fcst: {}})
+
+        #
+        # Get the availability interval of the current observation type from the
+        # verification configuration dictionary and use it to calculate the hours-
+        # of-day at which the obs will be available.
+        #
+        # Get the obs availability interval.
+        config_var_name = "".join([obtype, "_OBS_AVAIL_INTVL_HRS"])
+        obs_avail_intvl_hrs = vx_config[config_var_name]
+        # Ensure that the obs availability interval evenly divides into 24.
+        remainder = 24 % obs_avail_intvl_hrs
+        if remainder != 0:
+            msg = dedent(f"""
+                The obs availability interval for obs of type {obtype} must divide evenly
+                into 24 but doesn't:
+                  obs_avail_intvl_hrs = {obs_avail_intvl_hrs}
+                  24 % obs_avail_intvl_hrs = {remainder}"
+                """)
+            raise Exception(msg)
+        # Assume that the obs are available at hour 0 of the day regardless
+        # of obs type.
+        obs_avail_hr_start = 0
+        obs_avail_hr_end = obs_avail_hr_start + 24
+        # Construct list of obs availability hours-of-day.
+        obs_avail_hrs_of_day = [hr for hr in range(obs_avail_hr_start, obs_avail_hr_end, obs_avail_intvl_hrs)]
```

Comment:
Suggestion from pylint
```suggestion
        obs_avail_hrs_of_day = list(range(obs_avail_hr_start, obs_avail_hr_end, obs_avail_intvl_hrs))
```

Reply:
Changed.